x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/direct/DirectBlobContainerIndexInputTests.java

```diff
@@ -94,7 +96,9 @@ public int read(byte[] b, int off, int len) throws IOException {
                     };
                 }
             });
-        return new DirectBlobContainerIndexInput(blobContainer, fileInfo, newIOContext(random()), minimumReadSize,
+        final SearchableSnapshotDirectory directory = mock(SearchableSnapshotDirectory.class);
```

Comment:
As far as I can see, we pass the whole `SearchableSnapshotDirectory` into the `IndexInput` so that the `IndexInput` has access to the current time for stats purposes, leading to this mocking here -- otherwise we could continue to pass in just the `BlobContainer`. I think it would be neater to keep passing the `BlobContainer` and either pass in a `LongSupplier` for the current time, or else attach the current time supplier to the `IndexInputStats`.

Reply:
> attach the current time supplier to the IndexInputStats

I like this one, so I pushed 47d7e4c329d1a568352151b5490c0751717c316b